### PR TITLE
[FX-2184] Updates captions

### PR DESCRIPTION
**Pull request by dzucconi**
Re: [FX-2184](https://artsyproduct.atlassian.net/browse/FX-2184)

This got a bit into the weeds and uncovered a bug or two which I'll log with a separate ticket.

### CODE REVIEW

File: src/Components/Publishing/Sections/FullscreenViewer/ArtworkCaption.tsx [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+        color="black100"
+      />
+
+      <Button
+        variant="secondaryGray"
+        // TODO: event should be the correct type in palette
+        onClick={(event: React.MouseEvent<HTMLButtonElement, MouseEvent>) => {
+          // TODO: This isn't a real link and that's annoying
+          // <Button> doesn't support `as`, once it does, replace
+          // click handler this with: as={RouterLink} to={`/artwork/${section.slug}`}
```
Comment (dzucconi) [inline src/Components/Publishing/Sections/FullscreenViewer/ArtworkCaption.tsx:29]: I'd like to update `<Button>` to support this, and I will, but it's a bit of a project for the time being.

File: src/Components/Publishing/Sections/FullscreenViewer/FullscreenViewer.tsx [hunk 2]
```diff
@@ -40,12 +46,6 @@ export class FullscreenViewer extends Component<
     }
   }
 
-  handleKeydown = e => {
-    if (e.keyCode === 27) {
-      this.close(e)
-    }
-  }
```
Comment (dzucconi) [inline src/Components/Publishing/Sections/FullscreenViewer/FullscreenViewer.tsx:47]: `ModalBase` handles this

File: src/Components/Publishing/Sections/FullscreenViewer/Caption.tsx [hunk 3]
```diff
@@ -1,123 +1,83 @@
...
-import styled, { StyledFunction } from "styled-components"
-import { ArtworkCaption } from "../ArtworkCaption"
+import PropTypes from "prop-types"
+import { ArtworkCaption } from "./ArtworkCaption"
+import { Media } from "Utils/Responsive"
 
-interface CaptionProps extends React.HTMLProps<HTMLDivElement> {
-  section?: any
+interface CaptionProps {
+  section?: any // TODO: What am I?
```
Comment (eessex) [inline src/Components/Publishing/Sections/FullscreenViewer/Caption.tsx:8]: `SectionData`: https://github.com/artsy/reaction/blob/7bf92155d83cad72b6c24582bb79879294b5511e/src/Components/Publishing/Typings.ts#L129

Comment (eessex) [reply src/Components/Publishing/Sections/FullscreenViewer/Caption.tsx:8]: I think the section naming might actually be misleading, and this is actually representative of a single item in an image array, `article.sections[i].images` - https://github.com/artsy/reaction/blob/7bf92155d83cad72b6c24582bb79879294b5511e/src/Components/Publishing/Typings.ts#L144

**Comment by dzucconi** [comment]
This should be good to go — I'll merge in a bit unless there's any other feedback

File: src/Components/Publishing/Sections/FullscreenViewer/Caption.tsx (hunk 3 above)
Comment (eessex) [reply src/Components/Publishing/Sections/FullscreenViewer/Caption.tsx:8]: This could still be typed as `ImageData`, otherwise looks ok.

**Comment by artsyit** [comment]
:rocket: PR was released in [`v28.3.0`](https://github.com/artsy/reaction/releases/tag/v28.3.0) :rocket:

**Pull request merged**